financial-templates-lib/clients/ExpiringMultiPartyEventClient.js

```diff
@@ -94,7 +99,31 @@ class ExpiringMultiPartyEventClient {
         disputeSucceeded: event.returnValues.disputeSucceeded
       });
     }
-    this.lastBlockNumberSeen = currentBlockNumber;
+
+    // NewSponsor events mapped against PositionCreated events to determine size of new positions created.
+    const newSponsorEventsObj = await this.emp.getPastEvents("NewSponsor", {
+      fromBlock: this.lastBlockNumberSeen,
+      toBlock: currentBlockNumber
+    });
+    for (let event of newSponsorEventsObj) {
+      // Every transaction that emits a NewSponsor event must also emit a PositionCreated event.
+      const positionCreatedEventObj = await this.emp.getPastEvents("PositionCreated", {
+        fromBlock: event.blockNumber,
+        toBlock: event.blockNumber
+      });
+
+      this.newSponsorEvents.push({
+        transactionHash: event.transactionHash,
+        blockNumber: event.blockNumber,
+        sponsor: event.returnValues.sponsor,
+        collateralAmount: positionCreatedEventObj[0].returnValues.collateralAmount,
+        tokenAmount: positionCreatedEventObj[0].returnValues.tokenAmount
+      });
+    }
+
+    // Add 1 so that we do not double-count the actual last block number seen.
+    this.lastBlockNumberSeen = currentBlockNumber + 1;
```

Comment:
Nice find! Would it be too much trouble to change the name of this variable? The name just seems a little incorrect now.

Reply:
Easy to switch, changed to `firstBlockNumberToSearch`